bundles/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/internal/handler/BaseHomeThingHandler.java

```diff
@@ -61,4 +64,45 @@ protected void onSuccessfulOperation() {
             updateStatus(ThingStatus.ONLINE);
         }
     }
+
+    /**
+     * Method overridden to set the failedReconnectAttempts field.
+     */
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        switch (status) {
+            case ONLINE: {
+                failedReconnectAttempts = 0;
+                break;
+            }
+            case OFFLINE: {
+                if (statusDetail == ThingStatusDetail.COMMUNICATION_ERROR) {
+                    if (failedReconnectAttempts < Integer.MAX_VALUE) {
+                        failedReconnectAttempts++;
+                        break;
+                    }
+                }
+            }
+            default: {
+                failedReconnectAttempts = Integer.MAX_VALUE;
```

Comment:
Won't this be a problem is thing never reaches **ONLINE** before going into **OFFLINE** with **COMMUNICATION_ERROR**?

Reply:
No.

The new functionality is solely for an automatic **_re_**-connect process to (try to) reconnect the Thing after it has been online and then encountered a communication error.

This has nothing to do with, nor does it interfere with, the **_first_**-connect process, which is done in the normal way through the OH framework via the handler's `initialize()`, `dispose()`, and `bridgeStatusChanged()` methods.